config/routes.rb

```diff
@@ -1,6 +1,18 @@
 Rails.application.routes.draw do
   # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
 
+  namespace :api do
+    namespace :v1 do
+      resources :users
+      post '/auth/login', to: 'authentication#login'
+
+      resources :vehicles
+
+      resources :reservations
+      get '/all_reservations', to: 'reservations#all_reservations'
+    end
+  end
```

Comment:
- [ ] Nice routes and nice use of the `#resources` method.  However, you are generating more routes than needed.  

## Explanation

By using the `only:` keyword parameter, you can generate only the routes that you are using.

[Look](https://launchschool.com/books/demystifying_rails/read/routes_and_resources):

![image](https://user-images.githubusercontent.com/57421823/201431229-f7861d5e-3462-4897-b6fa-1242581bf06f.png)

## Example

For the vehicles controller, I can see that you are only using index, create, show, and destroy (no edit or update).

So:

```ruby
resources :vehicles, only: [:index, :create, :show, :destroy]
```

You can do the same process with the `:users` and `:reservations` routes 👍 

That way, your routes will be more consistent, and only the ones needed will be generated💪